src/main/scala/is/hail/io/bgen/BgenLoader.scala

```diff
@@ -22,18 +22,16 @@ object BgenLoader {
   def load(hc: HailContext, files: Array[String], sampleFile: Option[String] = None,
     tolerance: Double, nPartitions: Option[Int] = None): GenericDataset = {
     require(files.nonEmpty)
-    val samples = sampleFile.map(file => BgenLoader.readSampleFile(hc.hadoopConf, file))
+    val preIds = sampleFile.map(file => BgenLoader.readSampleFile(hc.hadoopConf, file))
       .getOrElse(BgenLoader.readSamples(hc.hadoopConf, files.head))
 
-    val duplicateIds = samples.duplicates().toArray
-    if (duplicateIds.nonEmpty) {
-      val n = duplicateIds.length
-      warn(
-        s"""found $n duplicate sample ${ plural(n, "ID") }
-           |  Duplicate IDs: @1""".stripMargin, duplicateIds)
+    val (sampleIds, duplicates) = mangle(preIds, "D" * _)
```

Comment:
You can do the poll, but I **strongly** feel `_N` is the best solution and "D*" should be avoided at all costs.